packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,295 @@
+import Chai, { expect } from 'chai';
+import { chaiRetryPlugin } from '../chai-retry-plugin/chai-retry-plugin';
+
+Chai.use(chaiRetryPlugin);
+
+describe('chai-retry-plugin', () => {
+    it('should retry a function that eventually succeeds', async () => {
+        let attempts = 0;
+
+        const funcToRetry = () => {
+            attempts++;
+            if (attempts < 3) {
+                throw new Error('Failed');
+            }
+            return 'Success';
+        };
+
+        await expect(funcToRetry).to.retry({ retries: 5, delay: 10 });
+        expect(attempts).to.equal(3);
+    });
+
+    it('should retry a function that always fails and throw an error', async () => {
+        let attempts = 0;
+
+        const funcToRetry = () => {
+            attempts++;
+            throw new Error('Failed');
+        };
+
+        try {
+            await expect(funcToRetry).to.retry({ retries: 3 });
+
+            throw new Error('This should not be called');
+        } catch (error) {
+            expect(error).to.be.an.instanceOf(Error);
```

Comment:
why do we need to check `instanceOf(Error)`?

Reply:
We missed that moment - probably if the user's function throws we should attach original message to the message thrown be `retry`

Reply:
@yurii-ve agree, it would be useful

Reply:
@yurii-ve fixed